### PR TITLE
feat: implement JSX `...` notation

### PR DESCRIPTION
**Pull request by eric-wieser**
This makes `<tag {...atts}>{...children}</tag>` work, and adds support in the delaborator.

The delaborator prints `<tag foo={1} {...attrs}><child />{...children}</tag>` in a suboptimal way, but it's better than it was before.

### CODE REVIEW

File: ProofWidgets/Data/Html.lean [hunk 1]
```diff
@@ -73,36 +75,79 @@ scoped syntax "<" ident jsxAttr* "/>" : jsxElement
...
 
-def transformTag (n m : Ident) (ns : Array Ident) (vs : Array (TSyntax `jsxAttrVal))
+/-- Sends `#[a, b, c]` to `` `(term| $a ++ $b ++ $c)``-/
+def joinArrays {m} [Monad m] [MonadRef m] [MonadQuotation m] (arr : Array Term) : m Term := do
+  if h : 0 < arr.size then
+    arr.foldlM (fun x xs => `($x ++ $xs)) arr[0] (start := 1)
+  else
+    `(#[])
+
+/-- Collapse adjacent `inl (_ : α)`s into a `β` using `f`. -/
```
Comment (Vtec234) [inline ProofWidgets/Data/Html.lean:93]:
```suggestion
/-- Collapse adjacent `inl (_ : α)`s into a `β` using `f`.
For example, `#[.inl a₁, .inl a₂, .inr b, .inl a₃] ↦ #[← f #[a₁, a₂], b, ← f #[a₃]]`. -/
```

File: ProofWidgets/Data/Html.lean [hunk 2]
```diff
@@ -73,36 +75,80 @@ scoped syntax "<" ident jsxAttr* "/>" : jsxElement
...
+scoped syntax "{..." term "}" : jsxChild
 /-- Interpolates an expression into a JSX literal -/
 scoped syntax "{" term "}" : jsxChild
 scoped syntax jsxElement   : jsxChild
 
 scoped syntax:max jsxElement : term
 
-def transformTag (n m : Ident) (ns : Array Ident) (vs : Array (TSyntax `jsxAttrVal))
+/-- Sends `#[a, b, c]` to `` `(term| $a ++ $b ++ $c)``-/
+def joinArrays {m} [Monad m] [MonadRef m] [MonadQuotation m] (arr : Array Term) : m Term := do
```
Comment (Vtec234) [inline ProofWidgets/Data/Html.lean:87]: Could we move the general functions such as `joinArrays`/`foldInlsM`/`delabListLiteral`/... to a `ProofWidgets.Util` module?

**Review by Vtec234 — APPROVED** [review]
Looks good, thank you!

File: ProofWidgets/Data/Html.lean (hunk 2 above)
Comment (eric-wieser) [reply ProofWidgets/Data/Html.lean]: I've moved the new ones in this PR.

File: ProofWidgets/Data/Html.lean [hunk 3]
```diff
@@ -9,11 +9,12 @@ import Lean.PrettyPrinter.Delaborator.Basic
 import Lean.Server.Rpc.Basic
 
 import ProofWidgets.Component.Basic
+import ProofWidgets.Util
 
 /-! We define a representation of HTML trees together with a JSX-like DSL for writing them. -/
 
 namespace ProofWidgets
-open Lean Server
+open Lean Server ProofWidgets.Util
```
Comment (Vtec234) [inline ProofWidgets/Data/Html.lean:17]:
```suggestion
open Lean Server Util
```

Comment (eric-wieser) [reply ProofWidgets/Data/Html.lean]: I avoided this because I didn't want to open `Lean.Util` or `Lean.Server.Util`

**Pull request merged**